tests/rptest/tests/schema_registry_test.py

```diff
@@ -75,7 +75,9 @@ def get_subject_name(sns: str, topic: str, field: MessageField,
 }
 
 schema1_def = '{"type":"record","name":"myrecord","fields":[{"name":"f1","type":"string"}]}'
-schema2_def = '{"type":"record","name":"myrecord","fields":[{"name":"f1","type":"string"},{"name":"f2","type":"string","default":"foo"}]}'
+# Schema 2 is only backwards compatible
+schema2_def = '{"type":"record","name":"myrecord","fields":[{"name":"f1","type":["null","string"]},{"name":"f2","type":"string","default":"foo"}]}'
```

Comment:
The change here simply adds `null`. I was under the impression that a schema is backwards compatible if it deletes or adds options.

Reply:
Ahh, I had to lookup avro spec. `null` here means optional https://avro.apache.org/docs/1.11.1/specification/

```
{
  "type": "record",
  "name": "LongList",
  "aliases": ["LinkedLongs"],                      // old name for this
  "fields" : [
    {"name": "value", "type": "long"},             // each element has a long
    {"name": "next", "type": ["null", "LongList"]} // optional next element
  ]
}

```

Reply:
That's why I had to make the restart test compatibility, none. It was appearing as both forward and backward, which is why existing tests didn't cover this case.